# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieStorage.h
# Commit: 5aa3b2f1b5ec
# Full Hash: 5aa3b2f1b5ec8a63fa04f80cf4aed4aac2c6ce22
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-07 21:34:02
# Regressor Bug: 1839918
# File Overlap Count: 6
# Description:
#   Bug 1839918 - Refactor "cookie-changed" notifications to use nsICookieNotification. r=extension-reviewers,cookie-reviewers,sessionstore-reviewers,bvandersloot,edgul,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D185300
# ==============================================================================

diff -r fecbf97cc8db -r 5aa3b2f1b5ec netwerk/cookie/CookieStorage.h
--- a/netwerk/cookie/CookieStorage.h	Thu Sep 07 08:22:27 2023 +0000
+++ b/netwerk/cookie/CookieStorage.h	Thu Sep 07 08:25:54 2023 +0000
@@ -8,6 +8,7 @@
 
 #include "CookieKey.h"
 
+#include "nsICookieNotification.h"
 #include "nsIObserver.h"
 #include "nsTHashtable.h"
 #include "nsWeakReference.h"
@@ -112,13 +113,19 @@
 
   void RemoveAll();
 
-  void NotifyChanged(nsISupports* aSubject, const char16_t* aData,
+  void NotifyChanged(nsISupports* aSubject,
+                     nsICookieNotification::Action aAction,
+                     const nsACString& aBaseDomain,
+                     dom::BrowsingContext* aBrowsingContext = nullptr,
+                     bool aIsThirdPartyCookie = false,
                      bool aOldCookieIsSession = false);
 
   void AddCookie(nsIConsoleReportCollector* aCRC, const nsACString& aBaseDomain,
                  const OriginAttributes& aOriginAttributes, Cookie* aCookie,
                  int64_t aCurrentTimeInUsec, nsIURI* aHostURI,
-                 const nsACString& aCookieHeader, bool aFromHttp);
+                 const nsACString& aCookieHeader, bool aFromHttp,
+                 dom::BrowsingContext* aBrowsingContext,
+                 bool aIsThirdPartyCookie);
 
   static void CreateOrUpdatePurgeList(nsCOMPtr<nsIArray>& aPurgedList,
                                       nsICookie* aCookie);
@@ -149,8 +156,7 @@
 
   virtual const char* NotificationTopic() const = 0;
 
-  virtual void NotifyChangedInternal(nsISupports* aSubject,
-                                     const char16_t* aData,
+  virtual void NotifyChangedInternal(nsICookieNotification* aSubject,
                                      bool aOldCookieIsSession) = 0;
 
   virtual void RemoveAllInternal() = 0;